### PR TITLE
Added build mechanism

### PR DESCRIPTION
**Pull request by anmolsachan**
Signed-off-by: Anmol Sachan <anmol13694@gmail.com>

### CODE REVIEW

**Comment by nimrod-becker** [comment]
Awesome PR, thanks @anmolsachan anmolsachan!

File: extras/operator/jsonnet/prometheus.libsonnet [hunk 1]
```diff
@@ -0,0 +1,32 @@
+local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
+
+{
+  _config+:: {
+    namespace: 'default',
+
+    prometheus+:: {
+      name: 'k8s',
```
Comment (jeniawhite) [inline extras/operator/jsonnet/prometheus.libsonnet]: @anmolsachan 
If I'm not mistaken the Prometheus configuration has a ```ruleSelector``` with ```matchLabels```.
This means that the user should be able to configure the name of the Prometheus deployment prior to generating the PrometheusRule CRD.
Taking the labels aside, I assume that users won't deep dive to change that value as they would expect to change it in the build configuration file example.libsonnet.
Am I missing something?

Comment (nimrod-becker) [reply extras/operator/jsonnet/prometheus.libsonnet]: @anmolsachan any update on the question?

Comment (anmolsachan) [reply extras/operator/jsonnet/prometheus.libsonnet]: @jeniawhite @nimrod-becker Apologies for the late reply, I somehow missed this.

> If I'm not mistaken the Prometheus configuration has a `ruleSelector` with `matchLabels`.
> This means that the user should be able to configure the name of the Prometheus deployment prior to generating the PrometheusRule CRD.
> Taking the labels aside, I assume that users won't deep dive to change that value as they would expect to change it in the build configuration file example.libsonnet.

I wrote this just as to follow the convention on kubernetes mixins (https://github.com/coreos/kube-prometheus/blob/f0afafdb084e91e35a0cdc6ba9a4c8e3f210712b/jsonnet/kube-prometheus/prometheus/prometheus.libsonnet#L21 ) from which this is inspired from . 
This is how it works according to my understanding: All the files contain the default values for configuration according to their deployment and there can be huge number of configurations. Putting everything in on configuration file is possible but would not separate the concerns and would be bulky. Thus every file has its own configurations. Now, whatever needs to be changed can be be put in the example.libsonnet and this configuration will override the default one. So If someone wishes to change it from example.libsonnet, its possible, they will just need to add following to it :
```
_config+:: {
    namespace: 'User_defined',
    prometheus+:: {
      name: 'User_defined',
    },
  },
```
Example : https://github.com/coreos/kube-prometheus/blob/f0afafdb084e91e35a0cdc6ba9a4c8e3f210712b/example.jsonnet#L11 for namespace override.

Comment (jeniawhite) [reply extras/operator/jsonnet/prometheus.libsonnet]: Alright, that's exactly what I've meant.
Seems to be fine.
Thank you.

**Pull request merged**